File: js/basics/2b_DLL_custom_implementation.js
class Node {
    constructor(data) {
      this.data = data;
      this.next = null;
      this.prev = null;
    }
  }

  insertAtHead(data) {
    const newNode = new Node(data);
    if (this.head) {
      this.head.prev = newNode;
      newNode.next = this.head;
    }
    this.head = newNode;
  }

File: c++/basics/2b_DLL_custom_implementation.cpp
class Node {
public:
    int data;
    Node* next;
    Node* prev;

    Node(int value) {
        data = value;
        next = NULL;
        prev = NULL;
    }
};

void insertAtHead(Node*& head, int value) {
    Node* newNode = new Node(value);
    if (head != NULL) {
        head->prev = newNode;
    }
    newNode->next = head;
    head = newNode;
}